sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -345,6 +345,109 @@ case class MakeInterval(
     )
 }
 
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(days, hours, mins, secs) - Make DayTimeIntervalType duration from days, hours, mins and secs.",
+  arguments = """
+    Arguments:
+      * days - the number of days, positive or negative
+      * hours - the number of hours, positive or negative
+      * mins - the number of minutes, positive or negative
+      * secs - the number of seconds with the fractional part in microsecond precision.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(1, 12, 30, 01.001001);
+       1 12:30:01.001001000
+      > SELECT _FUNC_(100, null, 3);
+       NULL
+  """,
+  since = "3.2.0",
+  group = "datetime_funcs")
+// scalastyle:on line.size.limit
+case class MakeDuration(
+    days: Expression,
+    hours: Expression,
+    mins: Expression,
+    secs: Expression,
+    failOnError: Boolean = SQLConf.get.ansiEnabled)
```

Comment:
Since ansi interval, should always be true? @MaxGekk

Reply:
Well, IMO, we should keep the consistency with other functions such as MakeInterval

Reply:
I don't see any reasons to support non-ansi behavior for ANSI intervals. Let's remove this parameter.